index.html

```diff
@@ -3,314 +3,423 @@
     // different because we've change the options object.
     updateTable();
   });
 
-  document.getElementById('sort-desc').addEventListener('click', function () {
+    document.getElementById('sort-desc').addEventListener('click', function () {
     // Disable the sort if 'desc' was previously selected, otherwise set the sort to 'desc'
     if (options.sorted === 'desc') {
       options.sorted = false;
+      options.page = 0;
     } else {
       options.sorted = 'desc';
+      options.page = 0;
+    }
+
+    updateTable();
+  });
+
+      document.getElementById('filter-No').addEventListener('click', function () {
+    // Disable the sort if 'desc' was previously selected, otherwise set the sort to 'desc'
+    if (options.filtered === 'no rest') {
+      options.filtered = false;
+      options.page = 0;
+    } else {
+      options.filtered = 'no rest';
+      options.page = 0;
+    }
+
+    updateTable();
+  });
+
+      document.getElementById('filter-Prison').addEventListener('click', function () {
+    // Disable the sort if 'desc' was previously selected, otherwise set the sort to 'desc'
+    if (options.filtered === 'prison') {
+      options.filtered = false;
+      options.page = 0;
+    } else {
+      options.filtered = 'prison';
+      options.page = 0;
     }
 
     updateTable();
   });
 
-  // This is where we'll need to create new event listeners for our filter and next/previous page buttons.
-  // They'll look very similar to the above, just listening to different elements, and perhaps performing
-  // different actions when clicked.
-</script>
+      document.getElementById('filter-Post').addEventListener('click', function () {
+    // Disable the sort if 'desc' was previously selected, otherwise set the sort to 'desc'
+    if (options.filtered === 'post') {
+      options.filtered = false;
+      options.page = 0;
+    } else {
+      options.filtered = 'post';
+      options.page = 0;
+    }
+
+    updateTable();
+  });
+
+    document.getElementById('filter-Parole').addEventListener('click', function () {
+    // Disable the sort if 'desc' was previously selected, otherwise set the sort to 'desc'
+      console.log('yo');
+    if (options.filtered === 'parole') {
+      options.filtered = false;
+      options.page = 0;
+    } else {
+      options.filtered = 'parole';
+      options.page = 0;
+    }
+
+    updateTable();
+  });
+
+      document.getElementById('filter-Probation').addEventListener('click', function () {
+    // Disable the sort if 'desc' was previously selected, otherwise set the sort to 'desc'
+    if (options.filtered === 'probation') {
+      options.filtered = false;
+      options.page = 0;
+    } else {
+      options.filtered = 'probation';
+      options.page = 0;
+    }
+
+    updateTable();
+  });
+//Is there a hide element option? Want to hide previous when page===0. On the right track below?
+      if (options.page===0){
```

Comment:
General comment: your code will be more resilient to changes if you avoid this pattern here (and above) of listing out all of the possibilities as if, else if statements (the more advanced way to do this is to use a [switch statement](http://www.w3schools.com/js/js_switch.asp), but this is frowned on for the same reason. What happens if you add a page 3? Or your data gets a new category ("Restrictions on parole subject to blah")? Suddenly things start breaking because you haven't explicitly accounted for that possibility.

So although we could get the below to work (we'd need to nest the whole thing in a function like `updatePageEvents()` that gets called at the same time `updateTable()` does; there's also an issue where `addEventListener` doesn't remove existing events, so our code would be getting called multiple times, and probably crash at some point), let's rework this so that we don't need to repeatedly change what the buttons do.

You get two event listeners. One for the previous page, and one for the next page.

Code like you don't know how many pages there are. You can count on 0 being the first page, but you don't know what the upper limit is. This is actually perfectly reasonable, as what if a filter is applied?

The only time it's okay to set `options.page` to a fixed number is when that number is 0 (the only page we know exists). Anything else should be `options.page = options.page + 1` or `options.page = options.page - 1`. (We can write this like `options.page += 1` and `options.page -= 1` to be fancy).

I like your approach to hiding one (or both) of the buttons. But each of the event listeners will need to determine when it's called whether it's appropriate to hide which buttons.

You'll probably start running into challenges related to the interaction of pagination and filter. Consider how much logic you could move from the event listeners themselves into `updateTable()`, where you could have the benefit of, say, knowing how many states have passed the filter.